src/webgpu/util/device_pool.ts

```diff
@@ -60,10 +60,17 @@ export class DevicePool {
       // (Hopefully if the device was lost, it has been reported by the time endErrorScopes()
       // has finished (or timed out). If not, it could cause a finite number of extra test
       // failures following this one (but should recover eventually).)
-      const lostReason = holder.lostReason;
-      if (lostReason !== undefined) {
-        // Fail the current test.
-        unreachable(`Device was lost: ${lostReason}`);
+      const lost = holder.lost;
+      if (lost !== undefined) {
+        if (lost.reason !== 'destroyed') {
+          // Fail the current test.
+          unreachable(`Device was unexpectedly lost: ${lost.message}`);
+        }
+        if (holder === this.defaultHolder) {
+          this.defaultHolder = 'uninitialized';
+        } else {
+          this.nonDefaultHolders.deleteByDevice(holder.device);
+        }
```

Comment:
see above, this can basically just assert `lostInfo === undefined` (but with different messages like "Device was unexpectedly destroyed during the test")

Reply:
Done